ui/app/src/components/RenameDashboardDialog/RenameDashboardDialog.tsx

```diff
@@ -0,0 +1,111 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { ChangeEvent, Dispatch, DispatchWithoutAction, useState } from 'react';
+import { Dialog, DialogTitle, DialogContent, DialogActions, Button, TextField, IconButton } from '@mui/material';
+import { DashboardResource } from '@perses-dev/core';
+import CloseIcon from 'mdi-material-ui/Close';
+import { dashboardDisplayName } from '@perses-dev/core/dist/utils/text';
+
+export interface RenameDashboardDialogProps {
+  open: boolean;
+  onClose: DispatchWithoutAction;
+  onConfirm: Dispatch<string>;
+  dashboard: DashboardResource;
+}
+
+/**
+ * Dialog used to rename a dashboard.
+ * @param props.open Define if the dialog should be opened or not.
+ * @param props.closeDialog Provides the function to close itself.
+ * @param props.onConfirm Action to perform when user confirmed.
+ * @param props.dashboard The dashboard resource to rename.
+ * @constructor
+ */
+const RenameDashboardDialog = (props: RenameDashboardDialogProps) => {
+  const { open, onClose, onConfirm, dashboard } = props;
+  const [name, setName] = useState<string>();
+  const [error, setError] = useState<string>();
+
+  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    if (!e.target.value) {
+      setError('Required');
+    } else if (!/^[a-zA-Z0-9_.:-]+$/.exec(e.target.value)) {
+      setError('Special characters allowed are _ . : -');
```

Comment:
Whatever message we use, let's have the same message here, in CreateDashboardDialog, and in CreateProjectDialog and DeleteProjectDialog (see this PR https://github.com/perses/perses/pull/916) 

`Allowed special characters are ` seems to be good 👍